src/main/java/com/google/sps/server/ScheduleCalendar.java

```diff
@@ -0,0 +1,250 @@
+import com.google.api.services.calendar.model.CalendarListEntry;
+import com.google.api.services.calendar.model.Event;
+import com.google.api.services.calendar.model.Events;
+import com.google.api.services.calendar.model.EventDateTime;
+import com.google.api.client.util.DateTime;
+
+import java.util.Date;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Arrays;
+import java.io.IOException;
+
+import com.google.sps.data.*;
+
+public class ScheduleCalendar {
+  private String calendarId;
+  private com.google.api.services.calendar.Calendar client;
+
+   /**
+   * Creates a new secondary calendar to put a class schedule on.
+   * 
+   * @param client the user's Google calendar
+   */
+  public ScheduleCalendar(com.google.api.services.calendar.Calendar client) throws IOException {
+    try {
+      this.client = client;
+
+      // Get the timezone of the user's primary calendar
+      CalendarListEntry calendarListEntry = client.calendarList().get("primary").execute();
+      String calendarTimeZone = calendarListEntry.getTimeZone();
+
+      // Create a new calendar
+      com.google.api.services.calendar.model.Calendar newCalendar = new com.google.api.services.calendar.model.Calendar();
+      newCalendar.setSummary("class schedule");
+      newCalendar.setTimeZone(calendarTimeZone);
+
+      // Insert the new calendar
+      com.google.api.services.calendar.model.Calendar createdCalendar = client.calendars().insert(newCalendar).execute();
+      this.calendarId = createdCalendar.getId(); 
+    } catch (IOException e) {
+      System.out.println("Couldn't create secondary calendar");
+    }
+  }
+
+  /**
+   * Puts schedule onto secondary calendar
+   * 
+   * @param schedule the schedule to add to the calendar 
+   * @param calendarId the id of the secondary calendar to put the schedule on
+   */
+  public void addSchedule(Schedule schedule) throws IOException {
+    try {
+      // get current date, so can add the courses to the appropriate days of the current month 
+      java.util.Calendar calendar = java.util.Calendar.getInstance();
+      int nextSun = getDateOfNextSunday(calendar);
+      int month = getCurrentMonth(calendar);
+      int year = getCurrentYear(calendar);
+
+      // for each course in the schedule, add an event to the calendar
+      Iterator<Course> courseIterator = schedule.getCourses().iterator();
+      while (courseIterator.hasNext()) {
+        Course currentCourse = courseIterator.next();
+        List<TimeRange> sectionTimes = currentCourse.getSections().get(0).getMeetingTimes();
+        for (int i = 0; i < sectionTimes.size(); i++) {
+          String startTime = calculateStartTime(sectionTimes.get(i), month, year, nextSun);
+          String endTime = calculateEndTime(sectionTimes.get(i), month, year, nextSun);
+          addEvent(currentCourse, startTime, endTime);
+        }
+      } 
```

Comment:
Same comment on using for-each instead of iterator here to improve code readability.